src/Microsoft.Health.Fhir.Core/Models/PartialDateTime.cs

```diff
@@ -292,7 +301,7 @@ public static PartialDateTime Parse(string s)
             catch (Exception ex) when (ex is ArgumentException)
             {
                 // The input value was parsed correctly but one of the value provided were out of range.
-                throw new FormatException("Input string was not in a correct format. At least one portion of a date was invalid or out of range.", ex);
+                throw new RequestNotValidException("Input string was not in a correct format. At least one portion of a date was invalid or out of range: " + ex.Message);
```

Comment:
If it do 
``` HTTP
/Observation?_lastUpdated=2013-04-03T15:30
```

This error is returned from the API call:
> Input string was not in a correct format. At least one portion of a date was invalid or out of range: The 'second' portion of a date must be specified if 'hour' and 'minute' are specified.\r\nParameter name: second

The "Parameter name: second" portion does not make sense from the user's context. Ideally, the error would provide the search parameter name, but we don't do that in other cases (but feel free to fix that)

Reply:
Fixed exception message